endpoints/openrtb2/auction.go

```diff
@@ -844,10 +844,11 @@ func (deps *endpointDeps) validateRequest(req *openrtb_ext.RequestWrapper, isAmp
 	}
 	var gpp gpplib.GppContainer
 	if req.BidRequest.Regs != nil && len(req.BidRequest.Regs.GPP) > 0 {
-		gpp, err = gpplib.Parse(req.BidRequest.Regs.GPP)
-		if err != nil {
+		var errs []error
+		gpp, errs = gpplib.Parse(req.BidRequest.Regs.GPP)
+		if len(errs) > 0 {
 			errL = append(errL, &errortypes.Warning{
-				Message:     fmt.Sprintf("GPP consent string is invalid and will be ignored. (%v)", err),
+				Message:     fmt.Sprintf("GPP consent string is invalid and will be ignored. (%v)", errs[0]),
```

Comment:
Following up on your question above, given that the logic here suggests that GPP parse errors are all labeled warnings, and the fact that we don't return from this particular function when non-fatal errors are found, should we append all `errs` to `errL` here in order to have a more descriptive output in debug mode?

Unless we change course and decide to classify some GPP errors as fatal from now on.

Reply:
> GPP parse errors are all labeled warnings

Correct. They are warnings.

> should we append all `errs`

We could, but it would be useful? Is the first error "good enough" to let the CMP know something is malfunctioning and to go investigate? Would a publisher want to see multiple `GPP consent string is invalid and will be ignored...` warnings returned?

Reply:
The team discussed at length and ultimately landed on your current implementation where we just return the first error since all of the errors that could possibly be returned from the GPP library `Parse` function are considered warnings that result in the entire string being ignored.

Reply:
Cool. Thank you for discussing. I didn't have a strong opinion on the matter.